### DISCUSSION TITLE
How to implement multi-dimensional independent variables?

### DISCUSSION THREAD

**Discussion by burakatakan** [jonathf/chaospy#400] [Q&A] [answered]
Dear all,
I just wanted to play around with chaospy. The problems, we are interested in are typically functions of 2-3 independent variables/ coordinates.
In the examples I could only find 1D examples (time).
Could you perhaps provide an example for fitting samples from  a function like:

y[:] =  sin (alpha * x[0, :] * exp(beta * x[1, :]) 

to obtain a surrogate model?

Thanks in advance!
Burak

**Comment by jonathf** [comment, marked as answer]
I agree that examples of multivariate coordinates are lacking.

Following the [quick tutorial](https://chaospy.readthedocs.io/en/master/user_guide/quick_tutorial.html), how about redefining the model solver. E.g. if you want a 2D mesh, you can do something like this:
```python
coordinates = numpy.meshgrid(numpy.linspace(0, 10, 300), numpy.linspace(30, 40, 300))

def model_solver(parameters, coordinates=coordinates):
    """Simple ordinary differential equation solver."""
    alpha, beta = parameters
    return numpy.sin(alpha * coordinates[0, :]) * numpy.exp(beta * coordinates[1, :])
```

**Reply by burakatakan** [reply]
Thanks a lot!
